src/cloudformation_cli_python_lib/resource.py

```diff
@@ -168,72 +167,78 @@ def test_entrypoint(
             msg = str(e)
         return ProgressEvent.failed(HandlerErrorCode.InternalFailure, msg)
 
+    @staticmethod
     def _parse_request(
-        self, event_data: MutableMapping[str, Any]
+        event_data: MutableMapping[str, Any]
     ) -> Tuple[
-        Tuple[Optional[SessionProxy], Optional[boto3.Session], boto3.Session],
-        BaseResourceHandlerRequest,
+        Tuple[Optional[SessionProxy], Optional[SessionProxy], SessionProxy],
         Action,
         MutableMapping[str, Any],
         HandlerRequest,
     ]:
         try:
             event = HandlerRequest.deserialize(event_data)
-            caller_creds = event.requestData.callerCredentials
-            provider_creds = event.requestData.providerCredentials
-            platform_creds = event.requestData.platformCredentials
-            request: BaseResourceHandlerRequest = UnmodelledRequest(
-                clientRequestToken=event.bearerToken,
-                desiredResourceState=event.requestData.resourceProperties,
-                previousResourceState=event.requestData.previousResourceProperties,
-                logicalResourceIdentifier=event.requestData.logicalResourceId,
-            ).to_modelled(self._model_cls)
-            caller_sess = _get_boto_session(caller_creds, event.region)
-            # No need to proxy as platform creds are required in the request
-            platform_sess = boto3.Session(
-                aws_access_key_id=platform_creds.accessKeyId,
-                aws_secret_access_key=platform_creds.secretAccessKey,
-                aws_session_token=platform_creds.sessionToken,
-            )
-            provider_sess = None
-            if provider_creds:
-                provider_sess = boto3.Session(
-                    aws_access_key_id=provider_creds.accessKeyId,
-                    aws_secret_access_key=provider_creds.secretAccessKey,
-                    aws_session_token=provider_creds.sessionToken,
-                )
+            platform_sess = _get_boto_session(event.requestData.platformCredentials)
+            caller_sess = _get_boto_session(event.requestData.callerCredentials)
+            provider_sess = _get_boto_session(event.requestData.providerCredentials)
+            # zero out credentials. this isn't so much to prevent targeted abuse,
+            # but to prevent accidental logging and re-use
+            event.requestData.platformCredentials = None
+            event.requestData.callerCredentials = None
+            event.requestData.providerCredentials = None
+            if platform_sess is None:
+                raise ValueError("No platform credentials")
```

Comment:
looks like the creds may are already being inadvertently re-used across invocations. In an end-to-end test, a cwl re-invoke raises `valueError`. 

Need to investigate a bit further, but we may need to be doing something like java's [refreshClient](https://github.com/aws-cloudformation/cloudformation-cli-java-plugin/blob/master/src/main/java/software/amazon/cloudformation/proxy/CallbackAdapter.java#L23-L28) operation.

Reply:
dug into this a bit deeper and it looks like the scheduler needs the credentials to be part of the request for re-invokes to work (looks to be the same in the java plugin). So if we're going to scrub these we'll need to stash them somewhere and inject them back into the `handler_request` for `reschedule_after_minutes`.